lib/Runtime/Base/ThreadContext.cpp

```diff
@@ -3581,7 +3581,13 @@ bool
 ThreadContext::RecordImplicitException()
 {
     AddImplicitCallFlags(Js::ImplicitCall_Exception);
-    return !IsDisableImplicitException();
+    if (IsDisableImplicitException())
+    {
+        return false;
+    }
+    bool hasImplicitCallHappened = IsDisableImplicitCall() && (GetImplicitCallFlags() & ~Js::ImplicitCall_Exception);
+    // Do not throw if any implicit call flag is set
```

Comment:
Can you add a more explanatory comment here? Something about the reason for not just disabling implicit exceptions everywhere that implicit calls are disabled?